prog.go

```diff
@@ -155,14 +155,20 @@ func (p *BPFProg) SetAttachTarget(attachProgFD int, attachFuncName string) error
 	return nil
 }
 
-// TODO: fix API to return error
-func (p *BPFProg) SetProgramType(progType BPFProgType) {
-	C.bpf_program__set_type(p.prog, C.enum_bpf_prog_type(int(progType)))
+func (p *BPFProg) SetProgramType(progType BPFProgType) error {
+	retC := C.bpf_program__set_type(p.prog, C.enum_bpf_prog_type(int(progType)))
```

Comment:
If the cast `C.enum_bpf_prog_type(progType)` is not allowed, cast it as `C.int` directly as the other examples.